### PR TITLE
Created TopicDescription Class [5991]

### PR DESCRIPTION
**Pull request by AlvaroGonzalezMolinillo**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/0/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/0/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/0/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/138/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/127/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/137/)

File: include/fastdds/dds/topic/TopicDescription.hpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * @file TopicDescription.hpp
+ */
+
+#ifndef _FASTDDS_TOPIC_DESCRIPTION_HPP_
+#define _FASTDDS_TOPIC_DESCRIPTION_HPP_
+
+#include <string>
+
+namespace eprosima {
+namespace fastdds {
```
Comment (LuisGP) [inline include/fastdds/dds/topic/TopicDescription.hpp]: Add `namespace dds`.

File: include/fastdds/dds/topic/TopicDescription.hpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+{
+protected:
+    TopicDescription(
+            const std::string& name,
+            const std::string& type_name)
+        : name_(name)
+        , type_name_(type_name)
+    {}
+
+    ~TopicDescription(){}
```
Comment (LuisGP) [inline include/fastdds/dds/topic/TopicDescription.hpp]:
```suggestion
    ~TopicDescription()
    {}
```

File: include/fastdds/dds/topic/TopicDescription.hpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+   virtual DomainParticipant* get_participant() const = 0;
+
+    std::string get_name() const
```
Comment (LuisGP) [inline include/fastdds/dds/topic/TopicDescription.hpp]: return `const std::string&`.

File: include/fastdds/dds/topic/TopicDescription.hpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        return name_;
+    }
+
+    std::string get_type_name() const
```
Comment (LuisGP) [inline include/fastdds/dds/topic/TopicDescription.hpp]: return `const std::string&`.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/256/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/233/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/255/)

File: include/fastdds/dds/topic/TopicDescription.hpp (hunk 1 above)
Comment (laura-eprosima) [reply include/fastdds/dds/topic/TopicDescription.hpp]: Done

Comment (laura-eprosima) [reply include/fastdds/dds/topic/TopicDescription.hpp]: Done

File: include/fastdds/dds/topic/TopicDescription.hpp (hunk 2 above)
Comment (laura-eprosima) [reply include/fastdds/dds/topic/TopicDescription.hpp]: Changed

File: include/fastdds/dds/topic/TopicDescription.hpp (hunk 3 above)
Comment (laura-eprosima) [reply include/fastdds/dds/topic/TopicDescription.hpp]: Changed

File: include/fastdds/dds/topic/TopicDescription.hpp (hunk 4 above)
Comment (laura-eprosima) [reply include/fastdds/dds/topic/TopicDescription.hpp]: Changed

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/346/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/319/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/344/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/503/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/452/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/500/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/527/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/470/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/523/)

File: include/fastdds/dds/topic/TopicDescription.hpp (hunk 3 above)
Comment (LuisGP) [reply include/fastdds/dds/topic/TopicDescription.hpp]: Please, add `const` to the returned type.

File: include/fastdds/dds/topic/TopicDescription.hpp (hunk 4 above)
Comment (LuisGP) [reply include/fastdds/dds/topic/TopicDescription.hpp]: Please, add `const` to the returned type.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/572/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/499/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/568/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/586/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/510/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/582/)

**Comment by raquelalvarezbanos** [comment]
Mac reports a failure but all tests pass. Merging this in.

**Pull request merged**